packages/devtools_app/test/network/network_request_inspector_test.dart

```diff
@@ -0,0 +1,106 @@
+import 'package:devtools_app/devtools_app.dart';
+import 'package:devtools_app/src/screens/network/network_request_inspector.dart';
+import 'package:devtools_test/devtools_test.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:vm_service/vm_service.dart';
+
+import 'utils/network_test_utils.dart';
+
+void main() {
+  group('NetworkRequestInspector', () {
+    late NetworkController controller;
+    SocketProfile socketProfile;
+    HttpProfile httpProfile;
+    FakeServiceManager fakeServiceManager;
+    late String _clipboardContents;
+
+    setUp(() {
+      setGlobal(IdeTheme, IdeTheme());
+      _clipboardContents = '';
+      socketProfile = loadSocketProfile();
+      httpProfile = loadHttpProfile();
+      fakeServiceManager = FakeServiceManager(
+        service: FakeServiceManager.createFakeService(
+          socketProfile: socketProfile,
+          httpProfile: httpProfile,
+        ),
+      );
+      setGlobal(ServiceConnectionManager, fakeServiceManager);
+      controller = NetworkController();
+      // This intercepts the Clipboard.setData SystemChannel message,
+      // and stores the contents that were (attempted) to be copied.
+      SystemChannels.platform.setMockMethodCallHandler((MethodCall call) {
+        switch (call.method) {
+          case 'Clipboard.setData':
+            _clipboardContents = call.arguments['text'];
+            break;
+          case 'Clipboard.getData':
+            return Future.value(<String, dynamic>{});
+          case 'Clipboard.hasStrings':
+            return Future.value(<String, dynamic>{'value': true});
+          default:
+            break;
+        }
+
+        return Future.value(true);
+      });
```

Comment:
since this is copied code from another test, put this into a shared file (perhaps test_utils.dart) so that it can be reused